internal/launcher/analyze.go

```diff
@@ -0,0 +1,86 @@
+package launcher
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/grafana/k6deps"
+	"go.k6.io/k6/cmd/state"
+)
+
+var (
+	errScriptNotFound     = errors.New("script not found")
+	errUnsupportedFeature = errors.New("not supported")
+)
+
+// analyze returns the dependencies for the command to be executed.
+// Presently, only the k6 input script or archive (if any) is passed to k6deps for scanning.
+// TODO: if k6 receives the input from stdin, it is not used for scanning because we don't know
+// if it is a script or an archive
+func analyze(gs *state.GlobalState, args []string) (k6deps.Dependencies, error) {
+	dopts := &k6deps.Options{
+		LookupEnv: func(key string) (string, bool) { v, ok := gs.Env[key]; return v, ok },
+	}
+
+	scriptname, hasScript := scriptNameFromArgs(args)
+	if !hasScript {
+		gs.Logger.
+			Debug("The command did not receive an input script.")
+		return nil, errScriptNotFound
+	}
+
+	if scriptname == "-" {
+		gs.Logger.
+			Debug("Test script provided by Stdin is not yet supported from Binary provisioning feature.")
+		return nil, errUnsupportedFeature
+	}
+
+	if _, err := gs.FS.Stat(scriptname); err != nil {
+		gs.Logger.
+			WithField("path", scriptname).
+			WithError(err).
+			Debug("The requested test script's file is not available on the file system.")
+		return nil, errScriptNotFound
+	}
+
+	if strings.HasSuffix(scriptname, ".tar") {
+		dopts.Archive.Name = scriptname
+	} else {
+		dopts.Script.Name = scriptname
+	}
+
+	return k6deps.Analyze(dopts)
+}
+
+// scriptNameFromArgs returns the file name passed as input and true if it's a valid script name
+func scriptNameFromArgs(args []string) (string, bool) {
+	// return early if no arguments passed
+	if len(args) == 0 {
+		return "", false
+	}
+
+	// search for a command that requires binary provisioning and then get the target script or archive
+	for i, arg := range args {
+		switch arg {
+		case "run", "archive", "inspect", "cloud":
```

Comment:
@codebien I realized this logic has a false positive `cloud login` that fails because there's no script associated with the command. I need to rework this logic. The more "corner" cases I find, the more I think we should explore the alternative integration of this logic into the root command as exemplified in this (now outdated) PR https://github.com/grafana/k6/pull/4689